src/card.rs

```diff
@@ -11,7 +11,13 @@ pub struct Card {
 
 impl Card {
     pub fn fight(&self, other: &Card) -> FightResult {
-        todo!()
+        let do_fight = ((self.health > other.damage), (other.health > self.damage));
+        match do_fight {
+            (s, o) if s && o => FightResult::Draw,
+            (s, _) if s => FightResult::Win,
+            (_, o) if o => FightResult::Loss,
+            _ => FightResult::Tie,
+        }
```

Comment:
Ooh this is a nice way of doing it, I didn't think of using the `if` then the discard. It might be easier to do `(true, false)` kind of syntax for the match arms, but it's not too bad :)